cmd/utils/flags.go

```diff
@@ -1057,16 +1062,19 @@ func setLes(ctx *cli.Context, cfg *ethconfig.Config) {
 
 // MakeDatabaseHandles raises out the number of allowed file handles per process
 // for Geth and returns half of the allowance to assign to the database.
-func MakeDatabaseHandles() int {
+func MakeDatabaseHandles(databaseHandles int) int {
 	limit, err := fdlimit.Maximum()
 	if err != nil {
 		Fatalf("Failed to retrieve file descriptor allowance: %v", err)
 	}
+	if databaseHandles < limit {
+		limit = databaseHandles
+	}
 	raised, err := fdlimit.Raise(uint64(limit))
 	if err != nil {
 		Fatalf("Failed to raise file descriptor allowance: %v", err)
 	}
-	return int(raised / 2) // Leave half for networking and other stuff
+	return int(raised)
```

Comment:
It should still return a half for database usage. The passed handlers should refer to handles for entire Geth, not only database.

Reply:
Ok, so regarding this comment and this [one](https://github.com/ethereum/go-ethereum/pull/24477#discussion_r815643261), I must do

```
cfg.DatabaseHandles = MakeDatabaseHandles(cfg.FdLimit)
```

`FdLimit`, being read by a `FdLimitFlag`. FdLimit being a new configuration option for the TOML/Command line (in place of my addition of databaseHandle). Is That Right?

Reply:
exactly.